### ISSUE TITLE
Rewrite reportIntrospectionError calls

### ISSUE THREAD

**Issue by NTillmann** [facebook/prepack#1898]
Once upon a time, Prepack didn't have a real error-handling story besides just throwing an exception.
From this time, there are still a ton of `reportIntrospectionError` calls in the code base.

They should all be rewritten, one by one (separate pull-requests, please) into something like the following.
```js
    let error = new CompilerDiagnostic("appropriate error message", referenceToSourceLocation, "PPxxxx", "FatalError");
    realm.handleError(error);
    throw new FatalError();
```

Where PPxxxx should be the next currently unused error code; check
https://github.com/facebook/prepack/wiki/Prepack-diagnostics
for all currently defined error code.
* Create a new wiki page with the PP error number as its name.
* Add a link to it from the page that lists all error numbers.

**Comment by chanakyabhardwajj** [comment]
>> Create a new wiki page with the PP error number as its name.

What's the best way to submit wiki edits? Once the (code) PR is approved/merged, do you also merge the wiki changes from the fork?

**Comment by NTillmann** [comment]
As far as I understand, the wiki is separate from version control.
So, first add a wiki entry to "claim" an error code number; maybe add "(for PR under review)" to the wiki title, and then submit pull request.

**Comment by chanakyabhardwajj** [comment]
>>  first add a wiki entry...

I apologize if I am mistaken, but I can not edit the wiki at all (as I am not part of the organization). That’s why I asked this question.

**Comment by hermanventer** [comment]
If that is the case, things are a bit harder. I suppose if you fork the code base, you can edit the forked wiki and point to it in the PR. Any other suggestions for a good workflow are most welcome.

**Comment by chanakyabhardwajj** [comment]
Would the following workflow be acceptable?
https://gist.github.com/larrybotha/10650410

**Comment by bernard-lin** [comment]
We could also host it on the site instead of on the wiki?

**Comment by chanakyabhardwajj** [comment]
That would (imho) simplify it for everyone in the future. However, I am happy to contribute whichever way suits you the best.

**Comment by hermanventer** [comment]
It probably would make sense to put all of our documentation in a source controlled folder.

For now, however, let's stick to the Wiki and let's keep talking about the best way to do this. For example, which GitHub projects are models that we should look to for guidance?
